src/internal.c

```diff
@@ -4648,8 +4648,15 @@ int DhAgree(WOLFSSL* ssl, DhKey* dhKey,
         void* ctx = wolfSSL_GetDhAgreeCtx(ssl);
 
         WOLFSSL_MSG("Calling DhAgree Callback Function");
+#ifdef AVR
+        unsigned int agreeSzInt = *agreeSz;
+        ret = ssl->ctx->DhAgreeCb(ssl, dhKey, priv, privSz,
```

Comment:
Perhaps it would make more sense to update the PK callback for `DhAgreeCb` to use `word32`?